install_cni_test.go

```diff
@@ -0,0 +1,188 @@
+package main_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+)
+
+var script = "k8s-install/scripts/install-cni.sh"
+
+var defaultEnvVars = map[string]string{
+	"KUBERNETES_SERVICE_HOST": "127.0.0.1",
+	"KUBERNETES_SERVICE_PORT": "8080",
+}
+
+// runCniScript will run install-cni.sh, and return the output of the command and an error
+func runCniScript() (string, error) {
+	result, err := exec.Command(script).CombinedOutput()
+	if err != nil {
+		return string(result), err
+	}
+	return string(result), nil
+}
+
+var _ = BeforeSuite(func() {
+	err := os.Setenv("SLEEP", "false")
+	if err != nil {
+		Fail("Failed to set env var 'SLEEP'")
+	}
+})
+
+var _ = Describe("install-cni.sh tests", func() {
+	BeforeEach(func() {
+		// Create the needed dirs.
+		err := os.MkdirAll("/host/opt/cni/bin", 755)
```

Comment:
I understand that we're almost always running this in a container so it's probably OK, but I always feel a bit icky seeing test code creating / deleting directories :)

I don't think anything needs to change here though.

Reply:
I agree, and we also have to run these tests as root.  It might be worth doing something to prevent people from not running the containerized tests if possible?